### ISSUE TITLE
clipy: Too complex type used in two find functions

### ISSUE THREAD

**Issue by wthrajat** [utxo-teleport/teleport-transactions#45]
In the file `src/wallet/api.rs`, there are 2 functions where `clippy` warns us about too complex types used. Those are:
```rust
pub fn find_incomplete_coinswaps(
        &self,
    ) -> Result<
        HashMap<
            Hash160,
            (
                Vec<(ListUnspentResultEntry, &IncomingSwapCoin)>,
                Vec<(ListUnspentResultEntry, &OutgoingSwapCoin)>,
            ),
        >,
        WalletError,
    > {...
```
and
```rust
pub fn find_live_contract_unspents(
        &self,
    ) -> Result<
        (
            Vec<(&IncomingSwapCoin, ListUnspentResultEntry)>,
            Vec<(&OutgoingSwapCoin, ListUnspentResultEntry)>,
        ),
        WalletError,
    > {...
```
### Clippy warnings:
```bash
warning: very complex type used. Consider factoring parts into `type` definitions
   --> src/wallet/wallet.rs:836:10
    |
836 |       ) -> Result<
    |  __________^
837 | |         HashMap<
838 | |             Hash160,
839 | |             (
...   |
844 | |         WalletError,
845 | |     > {
    | |_____^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity
    = note: `#[warn(clippy::type_complexity)]` on by default

warning: very complex type used. Consider factoring parts into `type` definitions
   --> src/wallet/wallet.rs:946:10
    |
946 |       ) -> Result<
    |  __________^
947 | |         (
948 | |             Vec<(&IncomingSwapCoin, ListUnspentResultEntry)>,
949 | |             Vec<(&OutgoingSwapCoin, ListUnspentResultEntry)>,
950 | |         ),
951 | |         WalletError,
952 | |     > {
    | |_____^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity
```
## How to reproduce?
Run `cargo clippy` to see the warnings.

**Comment by rajarshimaitra** [comment]
This one should be included in the pack too..

```bash
warning: this function has too many arguments (10/7)
   --> src/taker/routines.rs:231:1
    |
231 | / pub(crate) async fn send_proof_of_funding_and_init_next_hop(
232 | |     socket_reader: &mut BufReader<ReadHalf<'_>>,
233 | |     socket_writer: &mut WriteHalf<'_>,
234 | |     this_maker: &OfferAndAddress,
...   |
241 | |     hashvalue: Hash160,
242 | | ) -> Result<(ContractSigsAsRecvrAndSender, Vec<ScriptBuf>), TakerError> {
    | |_______________________________________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments
    = note: `#[warn(clippy::too_many_arguments)]` on by default
    ```

**Comment by rajarshimaitra** [comment]
# 1st type
> In the file src/wallet/wallet.rs, there are 2 functions where clippy warns us about too complex types used. Those are:

The function where this happens is only called once in `scripts/wallet.rs`. The `script` module is a artifact of the previous architecture. We intend to throw away `script` module eventually and repurpose those apis in different ways. 

So the easy fix here is just define a new type for this crazy returns and be done with it.

I would go like

```rust
type CrazyType = (
    Vec<(ListUnspentResultEntry, IncomingSwapCoin)>,
    Vec<(ListUnspentResultEntry, OutgoingSwapCoin)>,
);
```
And use `CrazyType` in the returns. 

Bonus: both the clippies can be solved by one type def. It could require a bit of auxiliary changes but is quite trivial.  

# 2nd type
> This one should be included in the pack too..

This one is a bit involved. 

Task is to categorically separate this arg list into different buckets.
```rust
pub(crate) async fn send_proof_of_funding_and_init_next_hop(
    socket_reader: &mut BufReader<ReadHalf<'_>>,
    socket_writer: &mut WriteHalf<'_>,
    this_maker: &OfferAndAddress,
    funding_tx_infos: &Vec<FundingTxInfo>,
    next_peer_multisig_pubkeys: &[PublicKey],
    next_peer_hashlock_pubkeys: &[PublicKey],
    next_maker_refund_locktime: u16,
    next_maker_fee_rate: u64,
    this_maker_contract_txes: &[Transaction],
    hashvalue: Hash160,
) -> Result<(ContractSigsAsRecvrAndSender, Vec<ScriptBuf>), TakerError> {
```

my picks:

```rust 

type Connection = (socket_reader, socket_writer);
type ThisMakerInfo = (this_maker, funding_txs_infos, this_maker_contract_txs);

// and a struct for the NextPeer infos.
struct NetxPeerInfo{
    next_peer_multisig_pubkeys: &[PublicKey],
    next_peer_hashlock_pubkeys: &[PublicKey],
    next_maker_refund_locktime: u16,
    next_maker_fee_rate: u64,
    }
```

This will reduce the arg list to 4 items `(connection, this_maker_info, next_peer_info, hashvalue)`.

**Comment by wthrajat** [comment]
Fixed by #61/#58
